ipam/ring/ring.go

```diff
@@ -282,29 +301,57 @@ func (es entries) merge(other entries, ourPeer mesh.PeerName) (result entries, u
 		case mine.Token > theirs.Token:
 			// insert, checking that a range owned by us hasn't been split
 			if previousOwner != nil && *previousOwner == ourPeer && theirs.Peer != ourPeer {
-				err = errEntryInMyRange(theirs)
-				return
+				// check we have no allocations in the range that got split
+				if checkEntryHasAllocations(theirs, mine) {
```

Comment:
This checks `theirs.Free < size`, i.e. it is checking if _they_ have any allocations in the range.
I think, to do this correctly, there needs to be some way to check the actual allocations.
Maybe pass in a function that will do that?

Reply:
Added a function that uses `space` to achieve `checkEntryHasAllocations` for an entry in ring.